src/libopenrave/configurationspecification.cpp

```diff
@@ -412,7 +412,7 @@ void ConfigurationSpecification::AddDerivativeGroups(int deriv, bool adddeltatim
     bool hasdeltatime = false;
     FOREACH(itgroup,_vgroups) {
         string replacename;
-        int offset = -1;
+        offset = -1; // we have just set `offset` to `GetDOF()`, shall it be reset to -1?
```

Comment:
offset is used below, so please still declare the variable and move `int offset = GetDOF();` below FOREACH.

Reply:
done here